bindings/python/utils.py

```diff
@@ -516,7 +516,62 @@ def infer_dimensions(meta, values, strict=True):
         missing_dims = dimnames.difference(dims.keys())
         raise CannotInferDimensionError(
             f"insufficient number of properties provided to infer dimensions: "
-            f"{missing_dims}"
+            f"{missing_dims} for metadata: {meta.uri}"
         )
 
     return dims
+
+
+def get_referred_instances(inst, include_meta=False):
+    """Return a set with all instances that are directly or indirectly
+    referred to by `inst`.
+
+    This function follows instances referred to by collections and via
+    properties of type 'ref'.  Transaction parents are not included,
```

Comment:
It is not clear for me what transaction parents mean in that context. Do we need that to clarify the goal of the function?

Reply:
I added it to improve the documentation and also document what this function doesn't do, since transactions also refer to other instances (in addition to collections, ref types and metadata). I think it is ok to keep - the reader can just ignore the sentence is he/she doesn't know what a transaction is.